crates/revm/revm-inspectors/src/tracing/builder/parity.rs

```diff
@@ -218,13 +238,153 @@ impl ParityTraceBuilder {
     pub fn into_transaction_traces(self) -> Vec<TransactionTrace> {
         self.into_transaction_traces_iter().collect()
     }
+
+    /// Creates a VM trace by walking over [CallTraceNode]s
+    ///
+    /// does not have the code fields filled in
+    pub fn vm_trace(&self) -> VmTrace {
+        match self.nodes.get(0) {
+            Some(current) => self.make_vm_trace(current),
+            None => VmTrace { code: Default::default(), ops: Vec::new() },
+        }
+    }
+
+    /// returns a VM trace without the code filled in
+    ///
+    /// iteratively creaters a VM trace by traversing an arena
+    fn make_vm_trace(&self, start: &CallTraceNode) -> VmTrace {
+        let mut child_idx_stack: Vec<usize> = Vec::with_capacity(self.nodes.len());
+        let mut sub_stack: VecDeque<Option<VmTrace>> = VecDeque::with_capacity(self.nodes.len());
+
+        let mut current = start;
+        let mut child_idx: usize = 0;
+
+        // finds the deepest nested calls of each call frame and fills them up bottom to top
+        let instructions = loop {
+            match current.children.get(child_idx) {
+                Some(child) => {
+                    child_idx_stack.push(child_idx + 1);
```

Comment:
why the `+1` here?

Reply:
ah to read the next child when it gets popped from the stack, im sure theres another way to do it, like popping then incrementing, but it seems like this way it implicitly handles the base case